service/frontend/dcRedirectionHandler_test.go

```diff
@@ -92,42 +85,33 @@ func (s *dcRedirectionHandlerSuite) SetupTest() {
 
 	s.currentClusterName = cluster.TestCurrentClusterName
 	s.alternativeClusterName = cluster.TestAlternativeClusterName
-
-	s.controller = gomock.NewController(s.T())
-	s.mockDomainCache = cache.NewMockDomainCache(s.controller)
-	s.mockClientBean = client.NewMockBean(s.controller)
-	s.mockRemoteFrontendClient = workflowservicetest.NewMockClient(s.controller)
-	s.mockFrontendHandler = NewMockWorkflowHandler(s.controller)
-	s.mockClientBean.EXPECT().GetRemoteFrontendClient(s.alternativeClusterName).Return(s.mockRemoteFrontendClient).AnyTimes()
-
-	s.logger = loggerimpl.NewDevelopmentForTest(s.Suite)
 	s.domainName = "some random domain name"
 	s.domainID = "some random domain ID"
-	s.config = NewConfig(dynamicconfig.NewCollection(dynamicconfig.NewNopClient(), s.logger), 0, false)
 
-	s.mockClusterMetadata = &mocks.ClusterMetadata{}
-	s.mockClusterMetadata.On("GetCurrentClusterName").Return(s.currentClusterName)
-	s.mockClusterMetadata.On("IsGlobalDomainEnabled").Return(true)
-	metricsClient := metrics.NewClient(tally.NoopScope, metrics.Frontend)
+	s.mockDCRedirectionPolicy = &MockDCRedirectionPolicy{}
 
-	s.mockArchivalMetadata = &archiver.MockArchivalMetadata{}
-	s.mockArchiverProvider = &provider.MockArchiverProvider{}
-	s.service = service.NewTestService(s.mockClusterMetadata, nil, metricsClient, s.mockClientBean, s.mockArchivalMetadata, s.mockArchiverProvider, nil)
+	s.controller = gomock.NewController(s.T())
+	s.mockResource = resource.NewTest(s.controller, metrics.Frontend)
+	s.logger = s.mockResource.GetLogger()
+	s.mockDomainCache = s.mockResource.DomainCache
+	s.mockClusterMetadata = s.mockResource.ClusterMetadata
+	s.mockRemoteFrontendClient = s.mockResource.RemoteFrontendClient
```

Comment:
plz remove these redundant vars

Reply:
Removed domainCache, the other two are used.